test/controllers/DeparturesControllerSpec.scala

```diff
@@ -189,69 +189,109 @@ class DeparturesControllerSpec
     "Accept" -> "application/xml"
   )
 
-  private val fakeValidXmlRequest = FakeRequest(
+  private def fakeValidXmlRequest(route: String, bearerToken: String) = FakeRequest(
     method = "POST",
-    uri = routes.DeparturesController.post().url,
-    headers = FakeHeaders(validHeaders ++ Seq(HeaderNames.AUTHORIZATION -> s"Bearer ${appConfig.eisBearerToken}")),
+    uri = route,
+    headers = FakeHeaders(validHeaders ++ Seq(HeaderNames.AUTHORIZATION -> s"Bearer $bearerToken")),
     body = CC015B)
 
-  private val fakeInvalidXmlRequest = FakeRequest(
+  private def fakeInvalidXmlRequest(route: String, bearerToken: String) = FakeRequest(
     method = "POST",
-    uri = routes.DeparturesController.post().url,
-    headers = FakeHeaders(invalidHeaders ++ Seq(HeaderNames.AUTHORIZATION -> s"Bearer ${appConfig.eisBearerToken}")),
+    uri = route,
+    headers = FakeHeaders(invalidHeaders ++ Seq(HeaderNames.AUTHORIZATION -> s"Bearer $bearerToken")),
     body = CC015B)
 
-  private val fakeUnauthorizedEmptyHeaderXmlRequest = FakeRequest(
+  private def fakeUnauthorizedEmptyHeaderXmlRequest(route: String) = FakeRequest(
     method = "POST",
-    uri = routes.DeparturesController.post().url,
+    uri = route,
     headers = FakeHeaders(validHeaders),
     body = CC015B)
 
-  private val fakeUnauthorizedEmptyHeaderValueXmlRequest = FakeRequest(
+  private def fakeUnauthorizedEmptyHeaderValueXmlRequest(route: String) = FakeRequest(
     method = "POST",
-    uri = routes.DeparturesController.post().url,
+    uri = route,
     headers = FakeHeaders(validHeaders ++ Seq(HeaderNames.AUTHORIZATION -> s"")),
     body = CC015B)
 
-  private val fakeUnauthorizedXmlRequest = FakeRequest(
+  private def fakeUnauthorizedXmlRequest(route: String) = FakeRequest(
     method = "POST",
-    uri = routes.DeparturesController.post().url,
+    uri = route,
     headers = FakeHeaders(validHeaders ++ Seq(HeaderNames.AUTHORIZATION -> s"Bearer 123")),
     body = CC015B)
 
-  "POST IE015 with valid headers and valid bearer token" - {
-    "should return 202 Accepted" in {
-      val result = controller.post()(fakeValidXmlRequest)
-      status(result) mustEqual ACCEPTED
+  "gbpost" - {
+    "POST IE015 with valid headers and valid bearer token" - {
+      "should return 202 Accepted" in {
+        val result = controller.gbpost()(fakeValidXmlRequest(routes.DeparturesController.gbpost().url, appConfig.eisgbBearerToken))
+        status(result) mustEqual ACCEPTED
+      }
     }
-  }
 
-  "POST IE015 with invalid headers and valid bearer token" - {
-    "should return 202 Accepted" in {
-      val result = controller.post()(fakeInvalidXmlRequest)
-      status(result) mustEqual BAD_REQUEST
+    "POST IE015 with invalid headers and valid bearer token" - {
+      "should return 202 Accepted" in {
```

Comment:
```suggestion
      "should return 400 BadRequest" in {
```
Please update the test description